.pre-commit-config.yaml

```diff
@@ -0,0 +1,14 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+  - repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v19.1.7
```

Comment:
The CI uses clang-format-16: https://github.com/NVIDIA/cuda-quantum/blob/d8930461128a9c564d636d7531d21f836eb1a59e/.github/workflows/repo_checks.yml#L118-L120